From 6d57e9078e880a3dd232d579f42ac437a8f1ef7b Mon Sep 17 00:00:00 2001
From: Duan Jiong <djduanjiong@gmail.com>
Date: Sat, 8 Sep 2012 16:32:28 +0000
Subject: [PATCH v5 002/164] etherdevice: introduce help function eth_zero_addr()

Git-commit: 6d57e9078e880a3dd232d579f42ac437a8f1ef7b
Patch-mainline: v3.7-rc5
Reference: fate#317388, bsc#909491
Target: sle11-sp4

a lot of code has either the memset or an inefficient copy
from a static array that contains the all-zeros Ethernet address.
Introduce help function eth_zero_addr() to fill an address with
all zeros, making the code clearer and allowing us to get rid of
some constant arrays.

Signed-off-by: Duan Jiong <djduanjiong@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
---
 include/linux/etherdevice.h |   11 +++++++++++
 1 file changed, 11 insertions(+)

Index: linux-3.0-SLE11-SP4/include/linux/etherdevice.h
===================================================================
--- linux-3.0-SLE11-SP4.orig/include/linux/etherdevice.h
+++ linux-3.0-SLE11-SP4/include/linux/etherdevice.h
@@ -161,6 +161,17 @@ static inline void eth_random_addr(u8 *a
 #define random_ether_addr(addr) eth_random_addr(addr)
 
 /**
+ * eth_zero_addr - Assign zero address
+ * @addr: Pointer to a six-byte array containing the Ethernet address
+ *
+ * Assign the zero address to the given address array.
+ */
+static inline void eth_zero_addr(u8 *addr)
+{
+	memset(addr, 0x00, ETH_ALEN);
+}
+
+/**
  * eth_hw_addr_random - Generate software assigned random Ethernet and
  * set device flag
  * @dev: pointer to net_device structure
